### PR TITLE
ImagePreview URL protocol validation

### PR DESCRIPTION
**Pull request by swmitra**
This PR is aimed to restrict/validate non-empty protocol for image preview URL. 

Ping @petetnt @abose @nethip for review.

### CODE REVIEW

File: src/extensions/default/QuickView/main.js [hunk 1]
```diff
@@ -474,7 +477,8 @@ define(function (require, exports, module) {
 
         // Determine whether or not this URL/path is likely to be an image.
         var parsed = PathUtils.parseUrl(tokenString);
-        var hasProtocol = parsed.protocol !== "";
+        // If the URL has a protocol, check if it's one of the supported protocols
+        var hasProtocol = parsed.protocol !== "" && validProtocols.indexOf(parsed.protocol.trim()) !== -1;
```
Comment (abose) [inline src/extensions/default/QuickView/main.js]: to lower case

Comment (swmitra) [inline src/extensions/default/QuickView/main.js]: Done. Good catch :+1:

File: src/extensions/default/QuickView/main.js [hunk 2]
```diff
@@ -474,7 +477,8 @@ define(function (require, exports, module) {
 
         // Determine whether or not this URL/path is likely to be an image.
         var parsed = PathUtils.parseUrl(tokenString);
-        var hasProtocol = parsed.protocol !== "";
+        // If the URL has a protocol, check if it's one of the supported protocols
+        var hasProtocol = parsed.protocol !== "" && validProtocols.indexOf(parsed.protocol.trim().toLowerCase()) !== -1;
```
Comment (petetnt) [inline src/extensions/default/QuickView/main.js]: Maybe something like `hasValidProtocol` would be a better name for the variable?

Is the empty check (`parsed.protocol !== ""`) still needed?

Comment (swmitra) [inline src/extensions/default/QuickView/main.js]: Yeah, that would have been a better name , but as this was an existing variable , I haven't renamed it. 
Secondly , having (parsed.protocol !== "") check will help in cutting down the array search in lot of cases like a relative path image url.

Comment (petetnt) [inline src/extensions/default/QuickView/main.js]: @swmitra yup, that makes sense.

**Comment by ficristo** [comment]
This seems to workaround a bug in the library:
`require("thirdparty/path-utils/path-utils").parseUrl("https://foo.bar.com:8080/baz/").protocol`
"https:"
`require("thirdparty/path-utils/path-utils").parseUrl("foo.bar.com:8080/baz/").protocol`
"foo.bar.com:"
`require("thirdparty/path-utils/path-utils").parseUrl("foo.bar.com/baz/").protocol`
""
I would expect an empty string for the second test too, where there is a port.

**Comment by nethip** [comment]
@swmitra Did you get a chance to address comment put in by @ficristo ?

**Comment by ficristo** [comment]
@nethip I've filed a bug at https://github.com/jblas/path-utils/issues/2
I am not sure if there is anything more to do here.

**Comment by nethip** [comment]
LGTM

**Comment by nethip** [comment]
Thanks @ficristo for filing a bug. 

And thanks @swmitra for this PR

**Pull request merged**